Interface.C

```diff
@@ -93,12 +93,44 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
 
             // Assign the (x,y,z) locations to the vertices
             for (int i = 0; i < faceCenters.size(); i++)
-            {
-                vertices[verticesIndex++] = faceCenters[i].x();
-                vertices[verticesIndex++] = faceCenters[i].y();
-                if(dim_ == 3)//out-of-plane axis = z
-                    vertices[verticesIndex++] = faceCenters[i].z();
+              for (unsigned int d = 0; d < dim_; ++d)
+                vertices[verticesIndex++] = faceCenters[i][d];
+
+#ifdef ADAPTER_DEBUG_MODE
+            // Check if we are in the right layer in case of preCICE dimension 2
```

Comment:
```suggestion
            // Check if we are in the right layer in case of preCICE dimension 2
            // If there is at least one node with a different z-coordinate, then the (2D) geometry is not on the xy-plane, as required.
```